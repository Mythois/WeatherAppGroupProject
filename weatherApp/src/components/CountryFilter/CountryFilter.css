.countryFilter {
  display: flex;
  padding: 1rem;
}

.countryButton {
  border: 1px solid var(--compBg);
  padding: 1rem;
  height: 3.5rem;
  width: 48%;
  border-radius: 20px;
  box-sizing: border-box;
  background-color: var(--compBg);
  text-align: center;
  cursor: pointer;
  font-weight: bold;
  transition: background-color 0.2s ease-in-out;
  color: black;
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

.countryButton.active,
.countryButton:hover {
  background-color: var(--activeBg);
  border: 1px solid var(--compBg);
  outline: none;
}

.countryButton:hover {
  font-size: large;
  transition: font-size 0.3s ease-in-out;
}

/* Media query for screens with a maximum width of 768px */
@media (max-width: 940px) {
  .countryFilter {
    display: flex;
    flex-wrap: wrap;
    padding: 1rem 1rem 0 1rem;
    justify-content: center;
  }

  .countryButton {
    border: none;
    padding: 0.5rem;
    height: 2.5rem;
    width: calc(25% - 0.5rem);
    max-width: calc(25% - 0.5rem);
    border-radius: 20px;
    box-sizing: border-box;
    background-color: var(--compBg);
    text-align: center;
    cursor: pointer;
    font-weight: bold;
    transition: background-color 0.2s ease-in-out;
    color: black;
    margin: 0.25rem;
    font-size: 0.9rem;
  }
}
