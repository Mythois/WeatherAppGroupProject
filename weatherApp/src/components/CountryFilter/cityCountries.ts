const cityCountries: { [key: string]: string } = {
    Oslo: 'Norway',
    Bergen: 'Norway',
    Stavanger: 'Norway',
    Trondheim: 'Norway',
    Stockholm: 'Sweden',
    KÃ¸benhavn: 'Denmark',
    Helsinki: 'Finland',
    Berlin: 'Germany',
    Paris: 'France',
    London: 'United Kingdom',
  };
  
export default cityCountries;
  