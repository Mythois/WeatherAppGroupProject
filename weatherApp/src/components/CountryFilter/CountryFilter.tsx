import React from 'react';
import './CountryFilter.css';

interface CountryFilterProps {
  countries: string[];
  selectedCountry: string | null;
  onSelectCountry: (country: string | null) => void;
}

const CountryFilter: React.FC<CountryFilterProps> = ({ countries, selectedCountry, onSelectCountry }) => {
  return (
    <div className="countryFilter">
      <button
        className={`countryButton ${selectedCountry === null ? 'active' : ''}`}
        onClick={() => onSelectCountry(null)}
      >
        All
      </button>
      {countries.map((country, index) => (
        <button
          key={index}
          className={`countryButton ${selectedCountry === country ? 'active' : ''}`}
          onClick={() => onSelectCountry(country)}
        >
          {country}
        </button>
      ))}
    </div>
  );
};

export default CountryFilter;
