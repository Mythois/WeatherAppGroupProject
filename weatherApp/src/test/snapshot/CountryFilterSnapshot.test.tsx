// CountryFilter.test.tsx
import { expect, it, vi } from 'vitest'
import { render, fireEvent } from '@testing-library/react'
import CountryFilter from '../../components/CountryFilter/CountryFilter'

it('renders correctly', () => {
  const countries = ['USA', 'Canada', 'UK']
  const selectedCountry = 'USA'
  const onSelectCountry = vi.fn()

  const result = render(
    <CountryFilter countries={countries} selectedCountry={selectedCountry} onSelectCountry={onSelectCountry} />,
  )

  // Assertions based on the rendered component
  expect(result.getByText('All')).toBeInTheDocument()
  expect(result.getByText('USA')).toBeInTheDocument()
  expect(result.getByText('Canada')).toBeInTheDocument()
  expect(result.getByText('UK')).toBeInTheDocument()

  // Snapshot comparison
  expect(result).toMatchSnapshot('src/test/snapshot/__snapshots__/CountryFilter.test.tsx.snap')
})

it('handles country selection correctly', () => {
  const countries = ['USA', 'Canada', 'UK']
  const selectedCountry = 'USA'
  const onSelectCountry = vi.fn()

  const result = render(
    <CountryFilter countries={countries} selectedCountry={selectedCountry} onSelectCountry={onSelectCountry} />,
  )

  fireEvent.click(result.getByText('Canada'))

  expect(onSelectCountry).toHaveBeenCalledWith('Canada')
})
