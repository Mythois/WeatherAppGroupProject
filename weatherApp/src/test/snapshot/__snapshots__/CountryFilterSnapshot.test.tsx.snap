// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly > src/test/snapshot/__snapshots__/CountryFilter.test.tsx.snap 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="countryFilter"
      >
        <button
          class="countryButton "
        >
          All
        </button>
        <button
          class="countryButton active"
        >
          USA
        </button>
        <button
          class="countryButton "
        >
          Canada
        </button>
        <button
          class="countryButton "
        >
          UK
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="countryFilter"
    >
      <button
        class="countryButton "
      >
        All
      </button>
      <button
        class="countryButton active"
      >
        USA
      </button>
      <button
        class="countryButton "
      >
        Canada
      </button>
      <button
        class="countryButton "
      >
        UK
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
